src/cleanlab_codex/validator.py

```diff
@@ -114,9 +117,11 @@ def validate(
                 - Additional keys from a [`ThresholdedTrustworthyRAGScore`](/codex/api/python/types.validator/#class-thresholdedtrustworthyragscore) dictionary: each corresponds to a [TrustworthyRAG](/tlm/api/python/utils.rag/#class-trustworthyrag) evaluation metric, and points to the score for this evaluation as well as a boolean `is_bad` flagging whether the score falls below the corresponding threshold.
         """
         scores, is_bad_response = self.detect(query, context, response, prompt, form_prompt)
+        metadata = {} if metadata is None else metadata
+        metadata = {**metadata, **scores}
```

Comment:
Would it make sense to guard the addition of `**scores` behind a flag in `Validator.__init__()`, such as `log_scores: bool = True`?
Still undecided on what the default should be, but this might help make the behavior more explicit.

Reply:
Yeah, makes sense.